<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Options</class>
 <widget class="QDialog" name="Options">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>635</width>
    <height>500</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QLabel" name="TextLabel1">
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>14</pointsize>
       <weight>50</weight>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string>XCA Options</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Default hash algorithm</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="hashBox" name="hashAlgo"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>PKCS12 encryption algorithm</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="pkcs12EncBox" name="pkcs12EncAlgo"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>String types</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="mbstring"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="pkcs12LegacyMACALG">
         <property name="text">
          <string>Export PKCS#12 with legacy MAC ALG (SHA1)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="suppress">
         <property name="text">
          <string>Suppress success messages</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="disableNetscape">
         <property name="text">
          <string>Disable legacy Netscape extensions</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="transDnEntries">
         <property name="text">
          <string>Translate established x509 terms (%1 -&gt; %2)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="onlyTokenHashes">
         <property name="toolTip">
          <string>The hashing functionality of the token is not used by XCA.
It may however honor a restricted hash-set propagated by the token.
Especially EC and DSA are only defined with SHA1 in the PKCS#11 specification.</string>
         </property>
         <property name="text">
          <string>Only use hashes supported by the token when signing with a token key</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="noColorize">
         <property name="text">
          <string>Don't colorize expired certificates</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Certificate expiry warning threshold</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLineEdit" name="cert_expiry_num"/>
         </item>
         <item row="0" column="3">
          <widget class="focusCombo" name="cert_expiry_unit"/>
         </item>
         <item row="1" column="3">
          <widget class="focusCombo" name="ical_expiry_unit"/>
         </item>
         <item row="1" column="2">
          <widget class="QLineEdit" name="ical_expiry_num"/>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Send vCalendar expiry reminder</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Serial number length</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="serial_len">
           <property name="suffix">
            <string> bit</string>
           </property>
           <property name="minimum">
            <number>8</number>
           </property>
           <property name="maximum">
            <number>256</number>
           </property>
           <property name="singleStep">
            <number>8</number>
           </property>
           <property name="value">
            <number>64</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>241</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Distinguished name</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Mandatory subject entries</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <layout class="QVBoxLayout">
            <item>
             <widget class="QComboBox" name="extDNobj"/>
            </item>
            <item>
             <widget class="QListWidget" name="extDNlist"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QPushButton" name="extDNadd">
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="extDNdel">
              <property name="text">
               <string>Delete</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Explicit subject entries</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <layout class="QVBoxLayout" name="_2">
            <item>
             <widget class="QComboBox" name="expDNobj"/>
            </item>
            <item>
             <widget class="QListWidget" name="expDNlist">
              <property name="dragDropMode">
               <enum>QAbstractItemView::InternalMove</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="adapt_explicit_subj">
              <property name="text">
               <string>Dynamically arrange explicit subject entries</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QPushButton" name="expDNadd">
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="expDNdel">
              <property name="text">
               <string>Delete</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="expDNdefault">
              <property name="text">
               <string>Default</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>18</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>PKCS#11 provider</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QListView" name="pkcs11List">
         <property name="dragEnabled">
          <bool>true</bool>
         </property>
         <property name="acceptDrops">
          <bool>true</bool>
         </property>
         <property name="dragDropOverwriteMode">
          <bool>false</bool>
         </property>
         <property name="dragDropMode">
          <enum>QAbstractItemView::InternalMove</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="movement">
          <enum>QListView::Snap</enum>
         </property>
         <property name="resizeMode">
          <enum>QListView::Adjust</enum>
         </property>
         <property name="uniformItemSizes">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QPushButton" name="addButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Add</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="removeButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Remove</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="searchPkcs11">
           <property name="text">
            <string>Search</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>77</width>
             <height>17</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>focusCombo</class>
   <extends>QComboBox</extends>
   <header>widgets/FocusCombo.h</header>
  </customwidget>
  <customwidget>
   <class>hashBox</class>
   <extends>QComboBox</extends>
   <header>widgets/hashBox.h</header>
  </customwidget>
  <customwidget>
   <class>pkcs12EncBox</class>
   <extends>QComboBox</extends>
   <header>widgets/pkcs12EncBox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>hashAlgo</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Options</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>495</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Options</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>453</x>
     <y>495</y>
    </hint>
    <hint type="destinationlabel">
     <x>194</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>noColorize</sender>
   <signal>toggled(bool)</signal>
   <receiver>cert_expiry_num</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>noColorize</sender>
   <signal>toggled(bool)</signal>
   <receiver>cert_expiry_unit</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>448</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
